packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetService.java

```diff
@@ -115,14 +118,14 @@ public List<T> getAssets(final Integer page, final Integer pageSize) throws IOEx
 	 * @param id The ID of the asset to delete
 	 * @throws IOException If there is an error deleting the asset
 	 */
-	public void deleteAsset(final UUID id) throws IOException {
+	public T deleteAsset(final UUID id) throws IOException {
 		final Optional<T> asset = getAsset(id);
 		if (asset.isEmpty()) {
-			return;
+			throw new NotFoundException("Asset not found: " + id.toString());
```

Comment:
I'm not 100% sure we want to be making this change here. I think that the `NotFoundException` is specifically a HTTP error 404 exception which is a little weird for this service to throw... especially if some other back end class talks to it, it would be unexpected (vs just returning the empty optional and letting the downstream deal with it).

At minimum we would probably want to add the `NotFoundException` to the method signature so anything downstream would know to expect that (because I wouldn't :D).  Same for `updateAsset` below.

Reply:
To add to this - by throwing a _different_ exception or returning empty optional our controller(s) can create a nice error message for the front end.  `NotFoundException` extends responseStatusException so the front end error would end up including that UUID which isn't overly user-helpful

Reply:
Reverted back to the way it was!